src/app/feature/calendar/calendar-grid.component.ts

```diff
@@ -1,27 +1,27 @@
-import {ChangeDetectionStrategy, Component, inject} from '@angular/core';
-import {NgOptimizedImage} from '@angular/common';
-import {CalendarCardFactoryDirective} from './cards/calendar-card-factory.directive';
-import {CalendarService} from './services/calendar.service';
-import {DialogModule} from '@angular/cdk/dialog';
-import {SocialProfileUrlPipe} from './pipes/social-profile-url.pipe';
-import {ShakeWhenDisabledDirective} from './directives/shake-on-click.directive';
-import {OpenDetailsDialogDirective} from './directives/open-details-dialog.directive';
+import { DialogModule } from '@angular/cdk/dialog';
+import { NgOptimizedImage } from '@angular/common';
+import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import { CalendarCardFactoryDirective } from './cards/calendar-card-factory.directive';
+import { OpenDetailsDialogDirective } from './directives/open-details-dialog.directive';
+import { ShakeWhenDisabledDirective } from './directives/shake-on-click.directive';
+import { SocialProfileUrlPipe } from './pipes/social-profile-url.pipe';
+import { CalendarService } from './services/calendar.service';
 
 @Component({
-    selector: 'xmas-calendar-grid',
-    imports: [
-        CalendarCardFactoryDirective,
-        DialogModule,
-        NgOptimizedImage,
-        OpenDetailsDialogDirective,
-        ShakeWhenDisabledDirective,
-        SocialProfileUrlPipe
-    ],
-    templateUrl: './calendar-grid.component.html',
-    styleUrl: './calendar-grid.component.css',
-    changeDetection: ChangeDetectionStrategy.OnPush
+  selector: 'xmas-calendar-grid',
+  imports: [
+    CalendarCardFactoryDirective,
+    DialogModule,
+    NgOptimizedImage,
+    OpenDetailsDialogDirective,
+    ShakeWhenDisabledDirective,
+    SocialProfileUrlPipe,
+  ],
+  templateUrl: './calendar-grid.component.html',
+  styleUrl: './calendar-grid.component.css',
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CalendarGridComponent {
   private readonly calendarService = inject(CalendarService);
-  cards = this.calendarService.cards;
+  protected cards = this.calendarService.cards;
```

Comment:
Maybe just like this:
```ts
  protected readonly cards = inject(CalendarService).cards;
```

You can avoid private members above public in this way

Reply:
@danielglejzner would disagree with you https://x.com/DanielGlejzner/status/1861420332352991507, that's why I did not change this.

Reply:
Thanks for sharing! I'll look at it later, if I learn something new I will reopen the thread

Reply:
Many different opinions on this one 😎 - I prefer to keep service private readonly